Doxyfile

```diff
@@ -56,7 +56,7 @@ PROJECT_LOGO           =
 # entered, it will be relative to the location where doxygen was started. If
 # left blank the current directory will be used.
 
-OUTPUT_DIRECTORY       = ../doc
+OUTPUT_DIRECTORY       = ./doxygen-docs
 
```

Comment:
Default location should encourage user/developer to generate docs outside the source tree.
```suggestion
OUTPUT_DIRECTORY       = ../doxygen-docs

```

Reply:
Consider someone using, for instance, the dev container. VS Code will mount the project folder into the container, so that everything inside the folder is shared between host and guest machines. Having documentation out of tree means that it won't be shared between machines, it will take up extra space on the container, and there's no guarantee the user has access to the parent directory in the first place (meaning they won't even be able to generate the docs).

I've added the folder to the `.gitignore` so it won't be committed. Is there another reason to not have docs inside the source folder? This is consistent with documentation tools in other languages like `rustdoc`.